### ISSUE TITLE
AssertionError with `null` in `accessible` filter

### ISSUE THREAD

**Issue by achave11-ucsc** [DataBiosphere/azul#6265]
This seems to be a new assertion introduced in as part of the changes of PR #6021 for issue #5999.

```
[ERROR]	2024-05-15T20:27:00.061Z	1ca69f26-6120-4a6d-a149-fcee75b80d1f	azul-service-anvildev	Caught exception for path /index/summary
Traceback (most recent call last):
  File "/var/task/chalice/app.py", line 1917, in _get_view_function_response
    response = view_function(**function_args)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/app.py", line 1299, in get_summary
    return app.repository_controller.summary(catalog=app.catalog,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_controller.py", line 110, in summary
    response = self.service.summary(catalog, filters)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_service.py", line 275, in summary
    aggs = dict(executor.map(summary, aggs_by_authority))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lang/lib/python3.11/concurrent/futures/_base.py", line 619, in result_iterator
    yield _result_or_cancel(fs.pop())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lang/lib/python3.11/concurrent/futures/_base.py", line 317, in _result_or_cancel
    return fut.result(timeout)
           ^^^^^^^^^^^^^^^^^^^
  File "/var/lang/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/var/lang/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/var/lang/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_service.py", line 270, in summary
    return entity_type, self._summary(catalog=catalog,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_service.py", line 302, in _summary
    request = chain.prepare_request(self.create_request(catalog, entity_type))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 136, in prepare_request
    request = self.inner.prepare_request(request)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 136, in prepare_request
    request = self.inner.prepare_request(request)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 136, in prepare_request
    request = self.inner.prepare_request(request)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 183, in prepare_request
    query = self.prepare_query()
            ^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 228, in prepare_query
    for field_path, relation_and_values in self.prepared_filters.items():
                                           ^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/caching.py", line 189, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
                                               ^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 195, in prepared_filters
    filters_json = self.filters.reify(self.plugin, limit_access=self._limit_access())
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/__init__.py", line 147, in reify
    assert False, accessible
           ^^^^^
AssertionError: {False, True, None}
```

[CW Logs](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2024-05-15T23*3a00*3a00.000Z~start~'2024-05-15T18*3a34*3a19.000Z~timeType~'ABSOLUTE~tz~'LOCAL~editorString~'fields*20*40timestamp*2c*20*60headers.x-forwarded-for*60*2c*20*40message*0a*23*20*7c*20filter*20*40requestId*20like*20*2fed7a2db8*7c88a1a125*7c6faa3d4f*7c5e6a2b3c*7c094b2367*7c9d95173b*7cc70bdcd8*7c196a45c6*7c64d7a0fc*7c33315576*7c06b2c6a4*7c31d8be29*7ccba78a9c*7c741fdf70*7c06f55cd3*7cf2fd9d4b*7c7511868c*7c59050cb1*7c84c61172*7cfd1c2850*7cfb42946e*7c127cde9c*7c8f5d38d3*7cae55d93*7c716c4850*7c80e76ae3*7cb7e3774b*7c7778e4e2*7c6b624685*7c8d76fbf9*7c3a7a7edb*7ca88479b7*7c379bf8ce*7cc87b952c*7c5004d039*7ce195f93b*7c7caddb02*7cc96e81ee*7c16d60cfb*7c45f8580c*7cbdfcf7a6*7c2bd835a3*7c3469e2f1*7ce187edb9*7cc60a36c0*7c2ceb6de1*7c63a5fdda*7c97e518a*7c35fc74fc*7c9a1dc7d1*7cf3d52833*7cb3a86afd*7c0a3cbf33*7ce6cfdc2c*7c4af55a53*7c0439a90c*7c79eecc7a*7c251ed361*7c55391547*7c145dfc58*7c8f1c15c9*7cc683e8fe*7c0b5866de*7c6de99574*7ce3134146*7cab52d5d9*7ce6b2e33c*7c514c414c*7cad266ef0*7c62a05e36*7c9849951e*7c353ea4e*7c855757ee*7c8409ca51*7c131572fb*7c4c362bec*7c8ba70489*7c99e916d9*7c1f8bde56*7c97ef709c*7c7a47912f*7c402c591a*7c18bfe981*7c07a7e65d*7c7d83a924*7cdb6c8a70*7cdf914c0d*7cc7a37caa*7c41831557*7cb8eacd1a*7ce361afb1*7c6c560de5*7cbf50d73f*7c23d5b358*7c9633abaa*7c6ee4d0f*7c153b53a9*7c9690f5ac*7cec7ef13c*7c9dbb3767*7c9f06769b*7c9c3f34df*7cbc9841b0*7c1ca69f26*7cf6a56e64*7cffd8cae8*7c2370117a*7c3ae7bf04*7c87ad498a*7cb664b9d9*7c1c0ae6f4*7c3c747ebe*7cbde390b7*7c7e8ee9ab*7c4e42d158*7cad4a74e3*7c2b37760e*7c34095d74*7cfac81199*7c9d59cee*7c4088edb6*7cc9606e33*7caa8fdc07*7cb240cc05*7cc9107306*7caf86496d*7c3854b6df*7ce725b889*7c00b03c33*2f*0a*7c*20filter*20*40requestId*20like*20*221ca69f26*22*0a*23*20*7c*20filter*20*40message*20like*20*2fReceived*7cAssertionError*2f*0a*23*20*7c*20parse*20*40requestId*20*22*2a-*22*20as*20rId*0a*23*20*7c*20parse*20*40message*20*22Received*20*2a*20request*20for*20*27*2a*27*2c*22*20as*20method*2c*20path*0a*23*20*7c*20parse*20*40message*20*22AssertionError*3a*20*2a*22*20as*20AssertionError*0a*23*20*7c*20parse*20*60headers.x-forwarded-for*60*20*22*2a*2c*2a*22*20as*20IP*2c*20_*0a*23*20*7c*20stats*20count*28*2a*29*20by*20IP*2c*20*60headers.user-agent*60*2c*20method*2c*20path*2c*20AssertionError*0a*7c*20sort*20*40timestamp*20asc*0a*23*20*7c*20display*20rId*0a*7c*20limit*20200~queryId~'ae171c969f4cd82-adab9943-4bf879e-7259b04b-d0cfdec48c713320d4f2fdcc~source~(~'*2faws*2flambda*2fazul-service-anvildev))$26tab$3Dlogs)

To reproduce, run…
```
curl -X 'GET' \
  'https://service.anvil.gi.ucsc.edu/index/summary?catalog=anvil&filters=%7B%22accessible%22%3A%7B%22is%22%3A%5Bfalse%2Ctrue%2Cnull%5D%7D%7D' \
  -H 'accept: application/json'
```

**Comment by hannes-ucsc** [comment]
`/index/datasets` endpoint also affected, so

```
 curl 'https://service.anvil.gi.ucsc.edu/index/datasets?catalog=anvil&filters=%7B%22accessible%22%3A%7B%22is%22%3A%5Bfalse%2Ctrue%2Cnull%5D%7D%7D'
 curl 'https://service.anvil.gi.ucsc.edu/index/summary?catalog=anvil&filters=%7B%22accessible%22%3A%7B%22is%22%3A%5Bfalse%2Ctrue%2Cnull%5D%7D%7D'
  ```
  
  `dev` is also affected.

**Comment by nadove-ucsc** [comment]
@hannes-ucsc: "We discussed supporting null [on slack](https://ucsc-gi.slack.com/archives/C705Y6G9Z/p1709935931408469) but I changed my mind. The user should get a 400 response when they pass in `null` as a value for the `accessible` filter."

We already handle the source ID field as a special case in `app.validate_filters` to only support the `is` relation. We can handle the `accessible` field at the same time and also validate that neither field is `null`.
